pkg/ipam/allocator.go

```diff
@@ -377,24 +382,35 @@ func (ipam *IPAM) StartExpirationTimer(ip net.IP, pool Pool, timeout time.Durati
 	ipam.allocatorMutex.Lock()
 	defer ipam.allocatorMutex.Unlock()
 
-	ipString := ip.String()
-	if _, ok := ipam.expirationTimers[ipString]; ok {
+	key := timerKey{ip: ip.String(), pool: pool}
+	if _, ok := ipam.expirationTimers[key]; ok {
 		return "", fmt.Errorf("expiration timer already registered")
 	}
 
 	allocationUUID := uuid.New().String()
-	ipam.expirationTimers[ipString] = allocationUUID
+	stop := make(chan struct{})
+	ipam.expirationTimers[key] = expirationTimer{
+		uuid: allocationUUID,
+		stop: stop,
+	}
 
-	go func(ip net.IP, allocationUUID string, timeout time.Duration) {
-		ipString := ip.String()
-		time.Sleep(timeout)
+	go func(key timerKey, ip net.IP, pool Pool, allocationUUID string, timeout time.Duration, stop <-chan struct{}) {
+		timer := time.NewTimerWithoutMaxDelay(timeout)
```

Comment:
Small heads up for backporting, on earlier branches this needs to be `NewTimer` and drop the `pkg/time` changes.